regen-network/regen-web
# feat : APP-363 another user purchased credits modal

## Description

https://regennetwork.atlassian.net/browse/APP-363

This PR also addresses https://regennetwork.atlassian.net/browse/APP-409

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


### How to test

https://deploy-preview-2500--regen-marketplace.netlify.app/projects/1

1. If a web2 user or a visiting user is trying to buy credits with USD and the selected credits are not available anymore and there is other sell orders with credits, the modal with copy "Sorry, another user has purchased all of the USD credits you selected!" should open -> https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=5596-91315&node-type=instance&m=dev

    - When clicking on the 'find another credit card project' button should take the suer to the projects page and filter projects showing credit card projects only (filter to be implemented)
    - When clicking on the 'or, buy with crypto' button:
        - If there is no connected wallet address, the connect wallet modal should open.
        - If the user is a web3 user but not connected to the wallet address associated to his/er account the switch wallet modal should open.

2. When trying to buy credits, if there are no credits available in all the sell orders, the modal with copy "Sorry, another user has purchased all of the available credits from this project" should open. -> https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=360-62730&node-type=instance&m=dev
    - When clicking on the 'search for new credits' button the user should be redirected to the projects page

3. If another user has bought some or all of the credits available to purchase, then the modal with copy "Sorry, another user has purchased some or all of the credits you selected!" should open. -> https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=179-109428&node-type=instance&m=dev
    - When clicking “choose new credits” the user should be redirected back to step 1 of the buying flow. If there are still credits available in the denom they selected, then it should still be selected win step 1. Otherwise, it should default again to USD or whatever denom is available.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
